src/actions/save/index.js

```diff
@@ -1,17 +1,46 @@
 /* @flow */
-
-type SaveActionHandlers = {|
+import { ValidationError } from "../../lib"
+type SaveActionConfig = {|
   // eslint-disable-next-line no-restricted-globals, promise/no-native
-  onCreateVaultToken: () => Promise<string>,
-  onApprove: ({| token: string |}) => void,
+  createVaultSetupToken: () => Promise<string>,
+  onApprove: ({| vaultSetupToken: string |}) => void,
 |};
 
-export type SaveAction = (SaveActionHandlers) => ({|
+export type SaveAction = (SaveActionConfig) => ({|
   type: "save",
-  ...SaveActionHandlers,
+  /* TODO: 
+      - We need to define how paymentSourceDetails is typed here
```

Comment:
A subsequent ticket will handle the definition of paymentSourceDetails. We'll type out the current expectation based on card fields data then, in the future, provide others as options.